src/SprykerEco/Zed/Computop/Communication/ComputopCommunicationFactory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Computop\Communication;
+
+use Generated\Shared\Transfer\MessageTransfer;
+use Spryker\Zed\Kernel\Communication\AbstractCommunicationFactory;
+use SprykerEco\Zed\Computop\ComputopDependencyProvider;
+
+/**
+ * @method \SprykerEco\Zed\Computop\ComputopConfig getConfig()
+ * @method \SprykerEco\Zed\Computop\Persistence\ComputopQueryContainerInterface getQueryContainer()
+ */
+class ComputopCommunicationFactory extends AbstractCommunicationFactory
+{
+
+    /**
+     * @return \SprykerEco\Zed\Computop\Dependency\Facade\ComputopToSalesInterface
+     */
+    public function getSalesFacade()
+    {
+        return $this->getProvidedDependency(
+            ComputopDependencyProvider::FACADE_SALES
+        );
+    }
+
+    /**
+     * @param int $orderId
+     *
+     * @return \Orm\Zed\Computop\Persistence\SpyPaymentComputop mixed
+     */
+    public function getComputopPaymentByOrderId($orderId)
+    {
+        return $this->getQueryContainer()->queryPaymentByOrderId($orderId)->findOne();
```

Comment:
A query in the communication factory is a code smell most likely. This should be moved down to the concrete class.